File: low-level-design-python/inmemorycachedesign/model/Cache.py
import collections


class Cache:
    def __init__(self, size, eviction_policy):
        self.size = size
        self.eviction_policy = eviction_policy
        self.data_store = collections.deque()
        self.cache_store = dict()

    def get_size(self):
        return self.size

    def get_eviction_policy(self):
        return self.eviction_policy

    def get_data_store(self):
        return self.data_store

    def get_cache_store(self):
        return self.cache_store

    def set_data_store(self, data_store):
        self.data_store = data_store

    def set_cache_store(self, cache_store):
        self.cache_store = cache_store

File: low-level-design-java/src/inmemorycachedesign/model/Cache.java
package inmemorycachedesign.model;

import inmemorycachedesign.constants.EvictionPolicy;

import java.util.HashMap;
import java.util.LinkedList;

public class Cache {
    private final int size;
    private final EvictionPolicy evictionPolicy;
    private LinkedList<Object> dataStore;
    private HashMap<Object, Object> cacheStore;

    public Cache(int size, EvictionPolicy evictionPolicy){
        this.size = size;
        this.evictionPolicy = evictionPolicy;
        this.dataStore = new LinkedList<>();
        this.cacheStore = new HashMap<>();
    }

    public int getSize(){
        return size;
    }

    public EvictionPolicy getEvictionPolicy(){
        return evictionPolicy;
    }

    public LinkedList<Object> getDataStore(){
        return dataStore;
    }

    public HashMap<Object, Object> getCacheStore(){
        return cacheStore;
    }

    public void setDataStore(LinkedList<Object> dataStore){
        this.dataStore = dataStore;
    }

    public void setCacheStore(HashMap<Object, Object> cacheStore){
        this.cacheStore = cacheStore;
    }

}
